client/src/game/comm/types/general.ts

```diff
@@ -67,3 +67,9 @@ export interface Note {
     text: string;
     uuid: string;
 }
+
+export interface Marker {
+    uuid: string;
+    name: string;
```

Comment:
I don't fully see the need for a separate name field here.

You have the uuid so you can always retrieve the shape name at any time.  If anything, storing the name in 2 different locations is bound for some sync issues.  I can already tell that at this point, renaming a shape will in fact not rename the marker until you refresh the page.

I think you want to keep the markers as a simple string[] array containing uuids in the current marker idea.

Reply:
This was kinda by design.... I thought that if you want to mark shapes with similar names you can differentiate them by naming them, marking them and then renaming them. You marker will then have a different, more distinctive name. But you're right, it doesn't have to be stored and might lead to confusion.

Reply:
On that note, if you want to make a more elaborate system where you want to 'share' makers etc, you might want to keep a name. And access-info (who has it), But realizing that, you might want to change the shape-focused marker to a name-to-shape focused marker so you can share the same shape with different marker-names....  tricky one. 

I'll remove it for now, it can always be added afterwards...

Reply:
And in doing so I just realized I wanted to keep them in an object to keep track of the two. In the menu on the left I display the name but keep track of the uuid. This way you can add markers with similar names but not with a duplicate uuid.

Reply:
This will be reset when you refresh though. If you want to keep them, you'll need to store them in the db as well.

Reply:
As you point out yourself, there is actually no mechanism to rename a marker and even if there was it would be overwritten on load by the shape name, unless you remove that aspect and set the default name for a new marker to the shape it is attached to.

You don't need to track the name in an object in order to display the name.  Just request the shape from the layermanager and show that name instead.


tl;dr  Either you offer true marker renaming where it is not associated with the shape itself or you only track the shape uuid as keeping track of the name on top of this is just bound to fail.

Reply:
That's what I don't get. Once it's pushed to array to show the name, I have no way of knowing from which shape-uuid it came unless I keep them together. Besides, once you click on the name, you want to know to which shape-uuid to go to. So you need to keep a relation between the two.

Reply:
`layerManager.UUIDMap.get(marker).name` will give you the name of the shape with uuid=marker